packages/aws-rfdk/lib/deadline/lib/sep-spotfleet.ts

```diff
@@ -0,0 +1,731 @@
+    const subnetIds = subnets.map(subnet => {
+      return subnet.subnetId;
+    });
+    const subnetId = subnetIds.length != 0 ? subnetIds.join(',') : undefined;
+
+    const instanceTagsToken = Lazy.any({ produce: () => {
+      return this.instanceTags.map(tag => {
+        return {
+          Key: tag.key,
+          Value: tag.value,
+        };
+      });
+    } });
+
+    const spotFleetRequestTags = Lazy.any({ produce: () => {
+      return this.spotFleetRequestTags.map(tag => {
+        return {
+          Key: tag.key,
+          Value: tag.value,
+        };
+      });
+    } });
+
+    let launchSpecifications: any[] = [];
+
+    props.instanceTypes.map(instanceType => {
+      const launchSpecification = {
+        BlockDeviceMappings: blockDeviceMappings,
+        IamInstanceProfile: {
+          Arn: iamProfile.attrArn,
+        },
+        ImageId: this.imageId,
+        KeyName: props.keyName,
+        SecurityGroups: securityGroupsToken,
+        SubnetId: subnetId,
+        TagSpecifications: [
+          {
+            ResourceType: ISpotFleetResourceType.INSTANCE,
+            Tags: instanceTagsToken,
+          },
+        ],
+        UserData: userDataToken,
+        InstanceType: instanceType.toString(),
+      };
+      launchSpecifications.push(launchSpecification);
+    });
+
+    const spotFleetRequestConfiguration = {
+      AllocationStrategy: props.allocationStrategy,
+      IamFleetRole: this.iamFleetRole.roleArn,
+      LaunchSpecifications: launchSpecifications,
+      ReplaceUnhealthyInstances: true,
+      TargetCapacity: props.targetCapacity,
+      TerminateInstancesWithExpiration: true,
+      Type: 'maintain',
+      ValidUntil: props.validUntil ? props.validUntil?.date.toUTCString() : undefined,
+      TagSpecifications: [
+        {
+          ResourceType: ISpotFleetResourceType.SPOT_FLEET_REQUEST,
+          Tags: spotFleetRequestTags,
+        },
+      ],
+    };
+
+    const sepSpotFleetRequestConfigurations = props.deadlineGroups.map(group => {
+      const sepSpotFleetRequestConfiguration = {
+        [group]: spotFleetRequestConfiguration,
+      };
+      return sepSpotFleetRequestConfiguration;
+    });
+
+    return sepSpotFleetRequestConfigurations;
+  }
+
+  private rfdkTagSpecification(): CfnTag {
+    const className = this.constructor.name;
+    const tagValue = `${RFDK_VERSION}:${className}`;
+    return {
+      key: TAG_NAME,
+      value: tagValue,
+    };
+  }
+
+  private validateProps(props: SEPSpotFleetProps) {
+    this.validateInstanceTypes(props.instanceTypes);
+    this.validateGroups(props.deadlineGroups);
+    this.validateArrayGroupsSyntax(props.deadlineGroups, /^(?!none$)[a-zA-Z0-9-_]+$/i, 'deadlineGroups');
+    this.validateRegion(props.deadlineRegion, /^(?!none$|all$|unrecognized$)[a-zA-Z0-9-_]+$/i);
+    this.validateBlockDevices(props.blockDevices);
+  }
+
+  private validateInstanceTypes(array: InstanceType[]) {
+    if (array.length == 0) {
```

Comment:
One of the requirements we had was:|
`They must use at least 5 different Spot instance types (drawn from the M,C,R,I,F,Z instance classes)`

Do we want to add here additional checks?